codec/dagjson/unmarshal.go

```diff
@@ -32,6 +32,10 @@ type DecodeOptions struct {
 	// If true, parse DAG-JSON `{"/":{"bytes":"base64 bytes..."}}` as a Bytes kind
 	// node rather than nested plain maps
 	ParseBytes bool
+
+	// If true, the decoder stops reading from the stream at the end of the JSON structure.
+	// I.e. it does not slurp remaining whitespaces and EOF.
```

Comment:
```suggestion
	// i.e. it does not slurp remaining whitespaces and EOF.
	// As per standard IPLD behavior, the parser considers the entire block to be
	// part of the JSON structure and will error if there is extraneous
	// non-whitespace data.
```

Something like that? I just want to make it clear that this is non-standard for IPLD, that it's appropriate to error if there's extraneous data and that this mode is for when you're using the parser for some other purpose than decoding discrete IPLD blocks.